sql/core/src/test/scala/org/apache/spark/sql/execution/adaptive/AdaptiveQueryExecSuite.scala

```diff
@@ -579,6 +579,155 @@ class AdaptiveQueryExecSuite
+        // The partition 0 in both left and right are all skewed.
+        // The partition 4 in left side is skewed.
+        // But for left outer join, we don't split the right partition even skewed.
+        // So the partition 0 in left side is divided into 5 splits(the max split number).
+        // the partition 4 in left side is divided into 5 splits(the max split number).
+        // So total (5 + 5 + 1) smjs.
+        // Union
+        // +- SortMergeJoin
+        //   +- Sort
+        //     +- CoalescedShuffleReader
+        //       +- ShuffleQueryStage
+        //   +- Sort
+        //     +- CoalescedShuffleReader
+        //       +- ShuffleQueryStage
+        // +- SortMergeJoin
+        //   +- Sort
+        //     +- SkewedShuffleReader
+        //       +- ShuffleQueryStage
+        //   +- Sort
+        //     +- SkewedShuffleReader
+        //       +- ShuffleQueryStage
+        //             .
+        //             .
+        //             .
+        // +- SortMergeJoin
+        //   +- Sort
+        //     +- SkewedShuffleReader
+        //       +- ShuffleQueryStage
+        //   +- Sort
+        //     +- SkewedShuffleReader
+        //       +- ShuffleQueryStage
+
+        val leftSmjAfter = findTopLevelSortMergeJoin(leftAdaptivePlan)
+        assert(leftSmjAfter.size == 11)
+
+        // skewed right outer join optimization
+        val (rightPlan, rightAdaptivePlan) = runAdaptiveAndVerifyResult(
+          "SELECT * FROM skewData1 right outer join skewData2 ON key1 = key2")
+        val rightSmj = findTopLevelSortMergeJoin(rightPlan)
+        assert(rightSmj.size == 1)
+        // left stats: [3496, 0, 0, 0, 4014]
+        // right stats:[6292, 0, 0, 0, 0]
+        // The partition 0 in both left and right side are all skewed.
+        // And the partition 4 in left side is skewed.
+        // But for right outer join, we don't split the left partition even skewed.
+        // And divide right side into 5 splits(the max split number)
+        // So total 6 smjs.
+        // Union
+        // +- SortMergeJoin
+        //   +- Sort
+        //     +- CoalescedShuffleReader
+        //       +- ShuffleQueryStage
+        //   +- Sort
+        //     +- CoalescedShuffleReader
+        //       +- ShuffleQueryStage
+        // +- SortMergeJoin
+        //   +- Sort
+        //     +- SkewedShuffleReader
+        //       +- ShuffleQueryStage
+        //   +- Sort
+        //     +- SkewedShuffleReader
+        //       +- ShuffleQueryStage
+        //             .
+        //             .
+        //             .
+        // +- SortMergeJoin
+        //   +- Sort
+        //     +- SkewedShuffleReader
+        //       +- ShuffleQueryStage
+        //   +- Sort
+        //     +- SkewedShuffleReader
+        //       +- ShuffleQueryStage
+
+        val rightSmjAfter = findTopLevelSortMergeJoin(rightAdaptivePlan)
+        assert(rightSmjAfter.size == 6)
+      }
+    }
+    spark.catalog.dropTempView("skewData1")
```

Comment:
nit: let's wrap the test with `withTempView`, which drops temp views automatically.